chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/watcher/outgoing/ChaosMonkeyRestTemplatePostProcessor.java

```diff
@@ -0,0 +1,25 @@
+package de.codecentric.spring.boot.chaos.monkey.watcher.outgoing;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.web.client.RestTemplate;
+
+/** @author Marcel Becker */
+public class ChaosMonkeyRestTemplatePostProcessor implements BeanPostProcessor {
+
+  private final ChaosMonkeyRestTemplateWatcher interceptor;
+
+  public ChaosMonkeyRestTemplatePostProcessor(final ChaosMonkeyRestTemplateWatcher interceptor) {
```

Comment:
Have you considered using a [RestTemplateCustomizer](https://www.baeldung.com/spring-rest-template-builder)? I think the is a preferred mechanism for modifying RestTemplate/WebClient

Reply:
Hi, I also thought about the customizer, but there is one big challenge/issue: Spring Boot just applies the RestTemplateCustomizers to RestTemplates which are created via injected RestTemplateBuilder. This approach will miss from my experience 99% of all created RestTemplate beans. 

The issue is also described very well in the article: https://medium.com/@TimvanBaarsen/spring-boot-why-you-should-always-use-the-resttemplatebuilder-to-create-a-resttemplate-instance-d5a44ebad9e9 

My suggestion would be to use the bean post processor approach in combination with the customizer.